### PR TITLE
feat: sitemap

### PR DESCRIPTION
**Pull request by TopETH**
## Summary by CodeRabbit

- **New Features**
  - Introduced new sitemaps for blog categories, projects, auctions, and presale pages to enhance SEO and navigation.

- **Bug Fixes**
  - Improved performance by advising against using arrow functions in certain components.

- **Refactor**
  - Consolidated and streamlined type declarations across multiple pages for better code organization and maintainability.

- **Chores**
  - Removed outdated dependencies related to Sanity from the project.
  
- **Documentation**
  - Updated `README.md` to remove outdated references and links.

### CODE REVIEW

**Review by gaboesquivel — CHANGES_REQUESTED** [review]
Thanks @TopETH, lgtm. Just a small change, please move shared and duplicated types to this file https://github.com/bitcashorg/smartsale/blob/develop/apps/webapp/types/routing.type.ts

File: apps/webapp/app/(routes)/[lang]/blog/[category]/sitemap.ts [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+  const {
+    params: { lang }
+  } = props
+  const sections = await getArticleSections(lang)
+  const categories = sections.map(section => section.slug)
+
+  return categories.map(category => ({
+    url: `${process.env.VERCEL_URL}/${lang}/blog/${category}`,
+    lastModified: new Date()
+  }))
```
[the remark this answers is not in the document]
Comment (gaboesquivel) [reply apps/webapp/app/(routes)/[lang]/blog/[category]/sitemap.ts:17]: @TopETH this makes sense

**Review by AndlerRL — CHANGES_REQUESTED** [review]
Most changes look good! I saw a couple of coderabbitai feedback, please take them a look and close them as resolved when finish :grin: 

Also, I notice the usage of `process.env.VERCEL_URL` which in simple view would work, but there could be a drawback:

@sheriffjimoh and I discovered that the `VERCEL_URL`, even defining it on the env vars, it won't update to the right URL on the environment where it was set (production, development, preview), and these come without the `https://` protocol if not set. So, to avoid potential errors, try to use instead `process.env.BASE_URL` and then define it on vercel env vars for more accuracy.

> There is a build error, can you please take a look? :eyes:

**Comment by TopETH** [comment]
> Thanks @TopETH, lgtm. Just a small change, please move shared and duplicated types to this file https://github.com/bitcashorg/smartsale/blob/develop/apps/webapp/types/routing.type.ts

Fixed

File: apps/webapp/app/(routes)/[lang]/blog/[category]/sitemap.ts (hunk 1 above)
[the remark this answers is not in the document]
Comment (TopETH) [reply apps/webapp/app/(routes)/[lang]/blog/[category]/sitemap.ts]: Fixed

**Comment by gaboesquivel** [comment]
image
@TopETH

**Pull request closed without merging**